src/main/java/org/alliancegenome/curation_api/services/Gff3Service.java

```diff
@@ -160,4 +174,74 @@
+			String uniqueId = Gff3UniqueIdHelper.getExonOrCodingSequenceUniqueId(gffEntry, attributes, dataProvider);
+			SearchResponse<Exon> response = exonDAO.findByField("uniqueId", uniqueId);
+			if (response == null || response.getSingleResult() == null) {
+				throw new ObjectValidationException(gffEntry, "uniqueId - " + ValidationConstants.INVALID_MESSAGE + " (" + uniqueId + ")");
+			}
+			Exon exon = response.getSingleResult();
+			
+			TranscriptExonAssociation transcriptAssociation = gff3DtoValidator.validateTranscriptExonAssociation(gffEntry, exon, attributes);
+			if (transcriptAssociation != null) {
+				idsAdded.get("TranscriptExonAssociation").add(transcriptAssociation.getId());
+				transcriptExonService.addAssociationToSubjectAndObject(transcriptAssociation);
+			}
+		} else if (StringUtils.equals(gffEntry.getType(), "CDS")) {
+			String uniqueId = Gff3UniqueIdHelper.getExonOrCodingSequenceUniqueId(gffEntry, attributes, dataProvider);
+			SearchResponse<CodingSequence> response = cdsDAO.findByField("uniqueId", uniqueId);
+			if (response == null || response.getSingleResult() == null) {
+				throw new ObjectValidationException(gffEntry, "uniqueId - " + ValidationConstants.INVALID_MESSAGE + " (" + uniqueId + ")");
+			}
+			CodingSequence cds = response.getSingleResult();
+			
+			TranscriptCodingSequenceAssociation transcriptAssociation = gff3DtoValidator.validateTranscriptCodingSequenceAssociation(gffEntry, cds, attributes);
+			if (transcriptAssociation != null) {
+				idsAdded.get("TranscriptCodingSequenceAssociation").add(transcriptAssociation.getId());
+				transcriptCdsService.addAssociationToSubjectAndObject(transcriptAssociation);
+			}
+		} else if (Gff3Constants.TRANSCRIPT_TYPES.contains(gffEntry.getType())) {
+			if (StringUtils.equals(gffEntry.getType(), "lnc_RNA")) {
+				gffEntry.setType("lncRNA");
+			}
+			if (!attributes.containsKey("ID")) {
+				throw new ObjectValidationException(gffEntry, "attributes - ID - " + ValidationConstants.REQUIRED_MESSAGE);
+			}
+			SearchResponse<Transcript> response = transcriptDAO.findByField("modInternalId", attributes.get("ID"));
+			if (response == null || response.getSingleResult() == null) {
+				throw new ObjectValidationException(gffEntry, "attributes - ID - " + ValidationConstants.INVALID_MESSAGE + " (" + attributes.get("ID") + ")");
+			}
+			Transcript transcript = response.getSingleResult();
+			
+			TranscriptGeneAssociation geneAssociation = gff3DtoValidator.validateTranscriptGeneAssociation(gffEntry, transcript, attributes, geneIdCurieMap);
+			if (geneAssociation != null) {
+				idsAdded.get("TranscriptGeneAssociation").add(geneAssociation.getId());
+				transcriptGeneService.addAssociationToSubjectAndObject(geneAssociation);
+			}
+		}
+		
+		return idsAdded;
+	}
+
+	public Map<String, String> getIdCurieMap(List<ImmutablePair<Gff3DTO, Map<String,String>>> gffData, BackendBulkDataProvider dataProvider) {
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>',' is not followed by whitespace.
<!-- __reviewdog__:ChBhMDBkODg5M2E5MTZjYzBkEgpjaGVja3N0eWxl -->